### ISSUE TITLE
Inconsistent OSX test failures.

### ISSUE THREAD

**Issue by DavidDeSimone** [Wilfred/remacs#514]
We are having `make check` for OSX periodically fail on Travis with the following error:

```
Test format-properties backtrace:
  signal(args-out-of-range (0 2))
  apply(signal (args-out-of-range (0 2)))
  (setq value-2 (apply fn-0 args-1))
  (unwind-protect (setq value-2 (apply fn-0 args-1)) (setq form-descri
  (if (unwind-protect (setq value-2 (apply fn-0 args-1)) (setq form-de
  (let (form-description-4) (if (unwind-protect (setq value-2 (apply f
  (let ((value-2 'ert-form-evaluation-aborted-3)) (let (form-descripti
  (let* ((fn-0 (function ert-equal-including-properties)) (args-1 (con
  (lambda nil (let* ((fn-0 (function ert-equal-including-properties)) 
  ert--run-test-internal(#s(ert--test-execution-info :test #s(ert-test
  ert-run-test(#s(ert-test :name format-properties :documentation nil 
  ert-run-or-rerun-test(#s(ert--stats :selector (not (tag :expensive-t
  ert-run-tests((not (tag :expensive-test)) #f(compiled-function (even
  ert-run-tests-batch((not (tag :expensive-test)))
  ert-run-tests-batch-and-exit((not (tag :expensive-test)))
  eval((ert-run-tests-batch-and-exit '(not (tag :expensive-test))))
  command-line-1(("-L" ":." "-l" "ert" "-l" "src/editfns-tests.el" "--
  command-line()
  normal-top-level()
Test format-properties condition:
    (args-out-of-range 0 2)
   FAILED   2/10  format-properties
```

Examples are https://github.com/Wilfred/remacs/pull/503 and https://github.com/Wilfred/remacs/pull/512. While developing on OSX, I am unable to reproduce this failure.

**Comment by shaleh** [comment]
Also, intermittent segfaults.

**Comment by shaleh** [comment]
https://blog.travis-ci.com/2017-11-21-xcode8-3-default-image-announce

Maybe relevant?

**Comment by brotzeit** [comment]
Seems to be fixed by #523.
